java/source/flight.rst

```diff
@@ -0,0 +1,1699 @@
+            AutoCloseables.close(holders.values());
+            holders.clear();
+        }
+    }
+
+
+    // ExampleFlightServer
+
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+
+    import java.io.IOException;
+
+    public class ExampleFlightServer implements AutoCloseable {
+        private final FlightServer flightServer;
+        private final Location location;
+        private final BufferAllocator allocator;
+        private final InMemoryStore mem;
+
+        /**
+         * Constructs a new instance using Allocator for allocating buffer storage that binds
+         * to the given location.
+         */
+        public ExampleFlightServer(BufferAllocator allocator, Location location) {
+            this.allocator = allocator.newChildAllocator("flight-server", 0, Long.MAX_VALUE);
+            this.location = location;
+            this.mem = new InMemoryStore(this.allocator, location);
+            this.flightServer = FlightServer.builder(allocator, location, mem).build();
+        }
+
+        public Location getLocation() {
+            return location;
+        }
+
+        public int getPort() {
+            return this.flightServer.getPort();
+        }
+
+        public void start() throws IOException {
+            flightServer.start();
+        }
+
+        public void awaitTermination() throws InterruptedException {
+            flightServer.awaitTermination();
+        }
+
+        public InMemoryStore getStore() {
+            return mem;
+        }
+
+        @Override
+        public void close() throws Exception {
+            AutoCloseables.close(mem, flightServer, allocator);
+        }
+    }
+
+    // Creating the Server
+
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+
+    BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+    ExampleFlightServer efs = new ExampleFlightServer(allocator, Location.forGrpcInsecure("localhost", 33333));
+    efs.start();
+
+    import org.apache.arrow.flight.FlightClient;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.memory.RootAllocator;
+
+    // client creation
+    RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE);
+    FlightClient client = FlightClient.builder(rootAllocator, Location.forGrpcInsecure("localhost", 33333)).build();
+
+    /**
+     * Lists flight information.
+     */
+    Iterable<FlightInfo> listFlights = client.listFlights(Criteria.ALL);
+
+    listFlights.forEach(t -> System.out.println(t));
+    System.out.println("Any list flight availale at this moment");
```

Comment:
This print seems a little confusing, maybe we can print before we print each flight and say something like "Available datasets:"?

Reply:
Deleted

Reply:
I don't see the change here.

Reply:
Updated